legate/core/store.py

```diff
@@ -950,6 +950,15 @@ def ndim(self) -> int:
         else:
             return self._shape.ndim
 
+    @property
+    def size(self) -> int:
+        s = 1
+        if self._ndim == 0:
+            return s
+        for p in self.shape:
+            s *= p
+        return s
```

Comment:
FYI `math.prod` was added in 3.8 
```
In [2]: math.prod((1,2,3))
Out[2]: 6

In [3]: math.prod(())
Out[3]: 1
```
(assuming shape is empty when ndim is zero)

Reply:
this is the code from when the minimum python version was 3.7. now we drop python 3.7 from the list of supported versions, we can and should replace this with `math.prod`. (and I suspect this isn't the only place where we roll a custom `prod` implementation.)

Reply:
Yes, this was basically copy-paste from cunumeirc. I have fixed it with the latest commit